libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -2090,7 +2090,9 @@ bool ExpressionCompiler::cleanupNeededForOp(Type::Category _type, Token::Value _
 {
 	if (Token::isCompareOp(_op) || Token::isShiftOp(_op))
 		return true;
-	else if (_type == Type::Category::Integer && (_op == Token::Div || _op == Token::Mod))
+	else if (_op == Token::Div || _op == Token::Mod)
+		return true;
+	else if (_type == Type::Category::FixedPoint && (_op == Token::Div || _op == Token::Mul || _op == Token::Mod))
```

Comment:
If the above is a generic `if div or mod` then those can be removed from here.

Reply:
@chriseth (I didn't write this code)